src/shared/install.c

```diff
@@ -2423,33 +2431,66 @@ int unit_file_query_preset(UnitFileScope scope, const char *root_dir, const char
                         if (strchr(COMMENTS, *l))
                                 continue;
 
-                        parameter = first_word(l, "enable");
-                        if (parameter) {
-                                if (fnmatch(parameter, name, FNM_NOESCAPE) == 0) {
-                                        log_debug("Preset file says enable %s.", name);
-                                        return 1;
-                                }
+                        if ((parameter = first_word(l, "enable"))) {
```

Comment:
Yeah... I waffled a bit on that one. I did it so that I could get an `else if` below. The flow was a little weird and the possibility of a leaked `state` was upsetting (though technically not possible).